lib/spack/spack/build_environment.py

```diff
@@ -1216,14 +1216,21 @@ def get_env_modifications(self) -> EnvironmentModifications:
         return env
 
     def _make_buildtime_detectable(self, dep: spack.spec.Spec, env: EnvironmentModifications):
-        if is_system_path(dep.prefix):
-            return
-
-        env.prepend_path("CMAKE_PREFIX_PATH", dep.prefix)
-        for d in ("lib", "lib64", "share"):
-            pcdir = os.path.join(dep.prefix, d, "pkgconfig")
-            if os.path.isdir(pcdir):
-                env.prepend_path("PKG_CONFIG_PATH", pcdir)
+        if not is_system_path(dep.prefix):
```

Comment:
Branch on external / not external, because `is_system_path` is incorrect for spack users on cernvm-fs

Reply:
If a package is external but not in a system path, I want to take this branch, so I could use some more detail to refine this: what happens on cernvm-fs?

Reply:
It's not rooted at `/`. You should let go of the concept of "system path", it does not exist.